### PR TITLE
Feat/extra tags

### PR DESCRIPTION
**Pull request by Rupalgw**
## what
* Add extra tag

## why
* Terraform Azure Modules only support tags from module.labels.tags,  if any user want to pass their custom tags

### CODE REVIEW

**Review by d4kverma — APPROVED** [review]
LGTM

File: main.tf [hunk 1]
```diff
@@ -193,7 +194,7 @@ resource "azurerm_storage_account" "storage" {
     for_each = var.cmk_encryption_enabled && var.identity_type != null ? [1] : []
     content {
       type         = var.identity_type
-      identity_ids = var.identity_type == "UserAssigned" ? [join("", azurerm_user_assigned_identity.identity.*.id)] : null
+      identity_ids = var.identity_type == "UserAssigned" ? [join("", azurerm_user_assigned_identity.identity[*].id)] : null
```
Comment (13archit) [inline main.tf:197]: pls remove join

File: main.tf [hunk 2]
```diff
@@ -276,7 +279,7 @@ resource "azurerm_key_vault_key" "kvkey" {
 ##-----------------------------------------------------------------------------
 resource "azurerm_storage_account_network_rules" "network-rules" {
   for_each                   = var.enabled ? { for rule in var.network_rules : rule.default_action => rule } : {}
-  storage_account_id         = join("", azurerm_storage_account.storage.*.id)
+  storage_account_id         = join("", azurerm_storage_account.storage[*].id)
```
Comment (13archit) [inline main.tf:282]: pls remove join

Comment (13archit) [reply main.tf:282]: and also replace [*] with [0]

File: main.tf [hunk 3]
```diff
@@ -307,7 +310,7 @@ resource "azurerm_key_vault_access_policy" "keyvault-access-policy" {
   count        = var.enabled && var.key_vault_rbac_auth_enabled == false ? 1 : 0
   key_vault_id = var.key_vault_id
   tenant_id    = data.azurerm_client_config.current.tenant_id
-  object_id    = join("", azurerm_user_assigned_identity.identity.*.principal_id)
+  object_id    = join("", azurerm_user_assigned_identity.identity[*].principal_id)
```
Comment (13archit) [inline main.tf:313]: remove join and can [*] be replaced with [0] ?

File: main.tf [hunk 4]
```diff
@@ -366,7 +369,7 @@ resource "azurerm_storage_share" "fileshare" {
 resource "azurerm_storage_table" "tables" {
   count                = var.enabled ? length(var.tables) : 0
   name                 = var.tables[count.index]
-  storage_account_name = join("", azurerm_storage_account.storage.*.name)
+  storage_account_name = join("", azurerm_storage_account.storage[*].name)
```
Comment (13archit) [inline main.tf:372]: Pls remove join and replace [*] with [0]

File: main.tf [hunk 5]
```diff
@@ -449,7 +452,7 @@ locals {
   resource_group_name   = var.resource_group_name
   location              = var.location
   valid_rg_name         = var.existing_private_dns_zone == null ? local.resource_group_name : (var.existing_private_dns_zone_resource_group_name == "" ? local.resource_group_name : var.existing_private_dns_zone_resource_group_name)
-  private_dns_zone_name = var.existing_private_dns_zone == null ? join("", azurerm_private_dns_zone.dnszone.*.name) : var.existing_private_dns_zone
+  private_dns_zone_name = var.existing_private_dns_zone == null ? join("", azurerm_private_dns_zone.dnszone[*].name) : var.existing_private_dns_zone
```
Comment (13archit) [inline main.tf:455]: Can we remove join function and replace [*] with [0]?

File: main.tf [hunk 6]
```diff
@@ -524,7 +527,7 @@ resource "azurerm_private_dns_zone_virtual_network_link" "addon_vent_link" {
   count                 = var.enabled && var.addon_vent_link ? 1 : 0
   name                  = format("%s-pdz-vnet-link-storage-addon", module.labels.id)
   resource_group_name   = var.addon_resource_group_name
-  private_dns_zone_name = var.existing_private_dns_zone == null ? join("", azurerm_private_dns_zone.dnszone.*.name) : var.existing_private_dns_zone
+  private_dns_zone_name = var.existing_private_dns_zone == null ? join("", azurerm_private_dns_zone.dnszone[*].name) : var.existing_private_dns_zone
```
Comment (13archit) [inline main.tf:530]: Can we remove join function and replace [*] with [0]?

File: main.tf [hunk 7]
```diff
@@ -534,11 +537,11 @@ resource "azurerm_private_dns_zone_virtual_network_link" "addon_vent_link" {
 ##-----------------------------------------------------------------------------
 resource "azurerm_private_dns_a_record" "arecord" {
   count               = var.enabled && var.enable_private_endpoint && var.diff_sub == false ? 1 : 0
-  name                = var.key_vault_id != null ? join("", azurerm_storage_account.storage.*.name) : null
+  name                = var.key_vault_id != null ? azurerm_storage_account.storage[*].name : null
```
Comment (13archit) [inline main.tf:540]: replace [*] with [0]

File: main.tf [hunk 8]
```diff
@@ -534,11 +537,11 @@ resource "azurerm_private_dns_zone_virtual_network_link" "addon_vent_link" {
...
   zone_name           = local.private_dns_zone_name
   resource_group_name = local.valid_rg_name
   ttl                 = 3600
-  records             = [data.azurerm_private_endpoint_connection.private-ip-0.0.private_service_connection.0.private_ip_address]
+  records             = [data.azurerm_private_endpoint_connection.private-ip-0[0].private_service_connection[0].private_ip_address]
```
Comment (13archit) [inline main.tf:544]: is it working fine?
did you test it?

File: outputs.tf [hunk 9]
```diff
@@ -1,41 +1,41 @@
 output "storage_account_id" {
-  value       = join("", azurerm_storage_account.storage.*.id)
+  value       = join("", azurerm_storage_account.storage[*].id)
```
Comment (13archit) [inline outputs.tf:2]: Remove join function and replace [*] with [0] wherever applicable. 
Pls do for all outputs.

File: main.tf [hunk 10]
```diff
@@ -307,7 +310,7 @@ resource "azurerm_key_vault_access_policy" "keyvault-access-policy" {
   count        = var.enabled && var.key_vault_rbac_auth_enabled == false ? 1 : 0
   key_vault_id = var.key_vault_id
   tenant_id    = data.azurerm_client_config.current.tenant_id
-  object_id    = join("", azurerm_user_assigned_identity.identity.*.principal_id)
+  object_id    = azurerm_user_assigned_identity.identity[*].principal_id
```
Comment (13archit) [inline main.tf:313]: replace [*] with [0]

File: main.tf [hunk 11]
```diff
@@ -449,7 +452,7 @@ locals {
   resource_group_name   = var.resource_group_name
   location              = var.location
   valid_rg_name         = var.existing_private_dns_zone == null ? local.resource_group_name : (var.existing_private_dns_zone_resource_group_name == "" ? local.resource_group_name : var.existing_private_dns_zone_resource_group_name)
-  private_dns_zone_name = var.existing_private_dns_zone == null ? join("", azurerm_private_dns_zone.dnszone.*.name) : var.existing_private_dns_zone
+  private_dns_zone_name = var.existing_private_dns_zone == null ? azurerm_private_dns_zone.dnszone[*].name : var.existing_private_dns_zone
```
Comment (13archit) [inline main.tf:455]: replace [*] with [0]

File: outputs.tf [hunk 12]
```diff
@@ -1,41 +1,41 @@
...
 
 output "storage_account_name" {
-  value       = join("", azurerm_storage_account.storage.*.name)
+  value       = azurerm_storage_account.storage[0].name
   description = "The name of the storage account."
 }
 
 output "storage_account_primary_location" {
-  value       = join("", azurerm_storage_account.storage.*.primary_location)
+  value       = azurerm_storage_account.storage[*].primary_location
```
Comment (13archit) [inline outputs.tf:12]: replace [*] to [0]

File: outputs.tf [hunk 13]
```diff
@@ -1,41 +1,41 @@
...
 
 output "storage_account_primary_web_endpoint" {
-  value       = join("", azurerm_storage_account.storage.*.primary_web_endpoint)
+  value       = azurerm_storage_account.storage[*].primary_web_endpoint
   description = "The endpoint URL for web storage in the primary location."
 }
 
 output "storage_account_primary_blob_endpoint" {
-  value       = join("", azurerm_storage_account.storage.*.primary_blob_endpoint)
+  value       = azurerm_storage_account.storage[*].primary_blob_endpoint
```
Comment (13archit) [inline outputs.tf:22]: replace [*] to [0]

File: outputs.tf [hunk 14]
```diff
@@ -1,41 +1,41 @@
...
   description = "The endpoint URL for blob storage in the primary location."
 }
 
 output "storage_account_primary_web_host" {
-  value       = join("", azurerm_storage_account.storage.*.primary_web_host)
+  value       =  azurerm_storage_account.storage[*].primary_web_host
```
Comment (13archit) [inline outputs.tf:27]: replace [*] to [0]

File: outputs.tf [hunk 15]
```diff
@@ -1,41 +1,41 @@
...
   description = "The hostname with port if applicable for web storage in the primary location."
 }
 
 output "storage_primary_connection_string" {
-  value       = join("", azurerm_storage_account.storage.*.primary_connection_string)
+  value       =  azurerm_storage_account.storage[*].primary_connection_string
```
Comment (13archit) [inline outputs.tf:32]: replace [*] to [0]

File: outputs.tf [hunk 16]
```diff
@@ -1,41 +1,41 @@
...
   sensitive   = true
   description = "The primary connection string for the storage account"
 }
 
 output "storage_primary_access_key" {
-  value       = join("", azurerm_storage_account.storage.*.primary_access_key)
+  value       = azurerm_storage_account.storage[*].primary_access_key
```
Comment (13archit) [inline outputs.tf:38]: replace [*] to [0]

File: main.tf (hunk 11 above)
Comment (13archit) [reply main.tf:455]: @Rupalgw

File: main.tf [hunk 17]
```diff
@@ -524,7 +527,7 @@ resource "azurerm_private_dns_zone_virtual_network_link" "addon_vent_link" {
   count                 = var.enabled && var.addon_vent_link ? 1 : 0
   name                  = format("%s-pdz-vnet-link-storage-addon", module.labels.id)
   resource_group_name   = var.addon_resource_group_name
-  private_dns_zone_name = var.existing_private_dns_zone == null ? join("", azurerm_private_dns_zone.dnszone.*.name) : var.existing_private_dns_zone
+  private_dns_zone_name = var.existing_private_dns_zone == null ? azurerm_private_dns_zone.dnszone[*].name : var.existing_private_dns_zone
```
[the remark this answers is not in the document]
Comment (13archit) [reply main.tf:530]: @Rupalgw

File: outputs.tf (hunk 16 above)
Comment (13archit) [reply outputs.tf:38]: @Rupalgw

File: outputs.tf [hunk 18]
```diff
@@ -1,41 +1,41 @@
...
 
 output "storage_account_primary_web_host" {
-  value       = join("", azurerm_storage_account.storage.*.primary_web_host)
+  value       = azurerm_storage_account.storage[*].primary_web_host
   description = "The hostname with port if applicable for web storage in the primary location."
 }
 
 output "storage_primary_connection_string" {
-  value       = join("", azurerm_storage_account.storage.*.primary_connection_string)
+  value       = azurerm_storage_account.storage[*].primary_connection_string
```
Comment (13archit) [inline outputs.tf:32]: Replace [*] to [0]

File: outputs.tf [hunk 19]
```diff
@@ -1,41 +1,41 @@
...
 
 output "storage_account_primary_blob_endpoint" {
-  value       = join("", azurerm_storage_account.storage.*.primary_blob_endpoint)
+  value       = azurerm_storage_account.storage[*].primary_blob_endpoint
   description = "The endpoint URL for blob storage in the primary location."
 }
 
 output "storage_account_primary_web_host" {
-  value       = join("", azurerm_storage_account.storage.*.primary_web_host)
+  value       = azurerm_storage_account.storage[*].primary_web_host
```
Comment (13archit) [inline outputs.tf:27]: Replace [*] to [0]

File: outputs.tf [hunk 20]
```diff
@@ -1,41 +1,41 @@
...
 
 output "storage_account_primary_location" {
-  value       = join("", azurerm_storage_account.storage.*.primary_location)
+  value       = azurerm_storage_account.storage[*].primary_location
   description = "The primary location of the storage account"
 }
 
 output "storage_account_primary_web_endpoint" {
-  value       = join("", azurerm_storage_account.storage.*.primary_web_endpoint)
+  value       = azurerm_storage_account.storage[*].primary_web_endpoint
```
[the remark this answers is not in the document]
Comment (13archit) [reply outputs.tf:17]: @Rupalgw

File: outputs.tf (hunk 12 above)
Comment (13archit) [reply outputs.tf:12]: @Rupalgw

**Review by 13archit — APPROVED** [review]

**Pull request merged**